processor.go

```diff
@@ -442,11 +443,133 @@ func callInterfaceFunc(handler, input interface{}, streaming bool) (intf interfa
 	return
 }
 
+// ProcessClientStreamRequest allows clients to push multiple messages
+// asynchronously to the same service with the same connection. Unlike in
+// ProcessClientRequest, we take a channel of inputs that can be filled and
+// will subsequently call the service with any new messages received in the
+// channel. The caller is responsible for closing the client input channel when
+// it is done.
+func (p *ServiceProcessor) ProcessClientStreamRequest(req *http.Request, path string,
+	clientInputs chan []byte) (chan []byte, error) {
+
+	outChan := make(chan []byte, 100)
+	var closeOutOnce sync.Once
+	mh, ok := p.handlers[path]
+
+	if !ok {
+		err := xerrors.New("the requested message hasn't been " +
+			"registered: " + path)
+		log.Error(err)
+		return nil, err
+	}
+
+	var stopServiceChan chan bool
+	var reply interface{}
+
+	// This goroutine listens on any new messages from the client and executes
+	// the request. Executing the request should fill the service's channel, as
+	// the service will use the same chanel for further requests.
+	go func() {
+		for {
+			select {
+			case buf, ok := <-clientInputs:
+				if !ok {
+					if stopServiceChan != nil {
+						close(stopServiceChan)
+					}
+					return
+				}
+
+				msg := reflect.New(mh.msgType).Interface()
+
+				err := protobuf.DecodeWithConstructors(buf, msg,
+					network.DefaultConstructors(p.Context.server.Suite()))
+				if err != nil {
+					log.Error(xerrors.Errorf("decoding: %v", err))
+					return
+				}
+
+				reply, stopServiceChan, err = callInterfaceFunc(mh.handler, msg, mh.streaming)
+				if err != nil {
+					log.Error(err)
+					if stopServiceChan != nil {
+						close(stopServiceChan)
+					}
```

Comment:
missing a return there ?

Reply:
Shouldn't the return be for any error ? Now it's returning only if the service returns a channel.

Reply:
ugh... absolutely!